### PR TITLE
Fix: example queued_interceptor_csrftoken.dart

### PR DESCRIPTION
**Pull request by seunghwanly**
Provide a more suitable example for queued_interceptor_csrftoken.dart

### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [x] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [x] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [ ] I have added the required tests to prove the fix/feature I'm adding
- [ ] I have updated the documentation (if necessary)
- [x] I have run the tests without failures
- [ ] I have updated the `CHANGELOG.md` in the corresponding package

### Additional context and info (if any)

This example adds 2 different `QueuedInterceptorsWrapper` and the last one handles `onError` to update csrf token.

### CODE REVIEW

File: example/lib/queued_interceptor_crsftoken.dart [hunk 1]
```diff
@@ -1,92 +1,95 @@
...
-  final dio = Dio();
-  //  dio instance to request token
-  final tokenDio = Dio();
-  String? csrfToken;
-  dio.options.baseUrl = 'https://seunghwanlytest.mocklab.io/';
-  tokenDio.options = dio.options;
+  String? cachedCSRFToken;
+
+  final dio = Dio()
+    ..options.baseUrl = 'https://httpbin.org/'
```
Comment (AlexV525) [inline example/lib/queued_interceptor_crsftoken.dart:13]: Use `httpbun.com` instead.

File: example/lib/queued_interceptor_crsftoken.dart [hunk 2]
```diff
@@ -1,92 +1,95 @@
-import 'dart:async';
-import 'dart:convert';
+import 'dart:developer';
```
Comment (AlexV525) [inline example/lib/queued_interceptor_crsftoken.dart:1]: Could you add the main idea and the overall flow at the top with comments?

File: example/lib/queued_interceptor_crsftoken.dart (hunk 1 above)
Comment (seunghwanly) [reply example/lib/queued_interceptor_crsftoken.dart:13]: OK, changed baseUrl and used [mix](https://httpbun.com/#mix) 👍

File: example/lib/queued_interceptor_crsftoken.dart (hunk 2 above)
Comment (seunghwanly) [reply example/lib/queued_interceptor_crsftoken.dart:1]: Yes, I added key features and scenario too

**Comment by seunghwanly** [comment]
Failed `test/test_suite_test.dart`
- [186: download cancels download (failed)](https://github.com/cfug/dio/actions/runs/8222507032/job/22484106342?pr=2128#step:15:186)
- [285: ignores zero duration timeouts (failed)](https://github.com/cfug/dio/actions/runs/8222507032/job/22484106342?pr=2128#step:15:286)

**Comment by seunghwanly** [comment]
I think there may be a syncing issue between my old fork repository and the current cfug/dio.
Closing this pr

**Pull request closed without merging**